### DISCUSSION TITLE
Cask: Homebrew asking for sudo to install cask

### DISCUSSION THREAD

**Discussion by SoCuul** [Homebrew/discussions#5328] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.20-14-gef1e365
ORIGIN: https://github.com/Homebrew/brew
HEAD: ef1e365e4399fc3a3d31bdbdead5c6423e626957
Last commit: 6 hours ago
Core tap HEAD: f8179fbd849b74bdf6d003a8988b76bace020a22
Core tap last commit: 13 minutes ago
Core tap JSON: 30 Apr 00:08 UTC
Core cask tap HEAD: 237d99eee733e51f2fe36d3738a4d19e05229b00
Core cask tap last commit: 31 minutes ago
Core cask tap JSON: 30 Apr 00:08 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.Zbbdx3Pdwb/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.44.0 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  python-mutagen
  python-brotli

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  cowsay
  libgit2@1.6
  lua@5.1
  openssl@1.1
```

### Description of issue

When installing some casks, homebrew prompts for sudo access to deal with the program. This only happens for some apps, but if I want to bulk update many apps the whole process will stop until I enter my password, which makes unattended usage painful.

Is there a way to get it to prompt at the start or skip over these packages until the end/not at all?

### Relevant casks

N/A

**Comment by gromgit** [comment]
SAFE: Manually cache your `sudo` credentials beforehand with `sudo -v`. You get 5 minutes before those credentials expire, so do your bulk update immediately thereafter.

RISKY: Write a script that outputs your sudo password, then `SUDO_ASKPASS=/path/to/script brew ...`. It's risky because anyone can use that script to authorise their own `sudo` usage with _your_ credentials.

**Reply by SoCuul** [reply]
> SAFE: Manually cache your `sudo` credentials beforehand with `sudo -v`. You get 5 minutes before those credentials expire, so do your bulk update immediately thereafter.
> 
> 
> 
> RISKY: Write a script that outputs your sudo password, then `SUDO_ASKPASS=/path/to/script brew ...`. It's risky because anyone can use that script to authorise their own `sudo` usage with _your_ credentials.

Would there be any way to cache it for longer? I'm not sure that 5 minutes is long enough.

**Reply by gromgit** [reply]
`man sudoers` and see the `timestamp_timeout` option. The `timestamp_type` option may also be of interest.

**Comment by SoCuul** [comment, marked as answer]
This gist pretty much gave me the exact solution I was looking for: https://gist.github.com/cowboy/3118588
